src/modules/governance/proposal/VotersListContainer.tsx

```diff
@@ -0,0 +1,139 @@
+import { Trans } from '@lingui/macro';
+import { Box, Button, CircularProgress, Typography } from '@mui/material';
+import fetch from 'isomorphic-unfetch';
+import { Fragment, useEffect, useState } from 'react';
+import { Row } from 'src/components/primitives/Row';
+import { CustomProposalType } from 'src/static-build/proposal';
+
+import { formatProposal } from '../utils/formatProposal';
+import { VotersListItem } from './VotersListItem';
+import { VotersListModal } from './VotersListModal';
+
+type VotersListProps = {
+  proposal: CustomProposalType;
+};
+
+// Possibly omit unnecessary values with lodash: proposalHistory, lastUpdateTimestamp, stkAave fields, votingHistory
+export type GovernanceVoter = {
+  aaveBalance: number; //2.6209877046554566
+  aavePropositionDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  aaveVotingDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  address: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  avatar: string | null; //null
+  handle: string | null; //null
+  isVerified: boolean; //false
+  lastUpdateTimestamp: number; //1601928545
+  name: string | null; //null
+  proposalHistory: unknown[]; //[]
+  propositionPower: number; //2.6209877046554566
+  propositionWeight: number; //1.6381173154096605e-7
+  stkAaveBalance: number; //0
+  stkAavePropositionDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  stkAaveVotingDelegate: string; //"0x8298a996a00835eedcc75763038b731fe617fd0d"
+  vote: 0 | 1; // 0 for nay, 1 for yae
+  votingHistory: unknown[]; //(2) [{…}, {…}]
+  votingPower: number; // the amount of voting power the user has - 2.6209877046554566
+  votingWeight: number; // the % that a single user contributes to the total - 1.6381173154096605e-7
+  _id: string; //mongodb objectid
+};
+
+export type VotersData =
+  | { yaes: GovernanceVoter[]; nays: GovernanceVoter[]; combined: GovernanceVoter[] }
+  | undefined;
+
+export const VotersListContainer = (props: VotersListProps): JSX.Element => {
+  const { proposal } = props;
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<null | string>(null);
+  const [voters, setVoters] = useState<VotersData>();
+  const [votersModalOpen, setVotersModalOpen] = useState(false);
+  const isMobile = false;
+
+  const votersUrl = `${process.env.NEXT_PUBLIC_API_BASEURL}/data/proposal-top-voters`;
```

Comment:
why add centralization here? You already have all the votes cached in build cache and can fetch the missing ones via events.

Reply:
For this first version, there really isn't a strong case for using the endpoint vs build cache besides caching ENS data, however there are plans to expand this into a full dashboard with user profiles and this endpoint allows for:
- Storing, ENS and Twitter data (via [sybil](https://raw.githubusercontent.com/Uniswap/sybil-list/master/verified.json))
- Storing current voting/proposition power for each user, including delegations
- Storing voting/proposition history for each user